packages/core-data/src/index.js

```diff
@@ -21,20 +26,32 @@ const createEntityRecordSelector = ( source ) => defaultEntities.reduce( ( resul
 	return result;
 }, {} );
 
-const createEntityRecordResolver = ( source ) => defaultEntities.reduce( ( result, entity ) => {
+const createEntityRecordAction = ( source ) => defaultEntities.reduce( ( result, entity ) => {
 	const { kind, name } = entity;
 	result[ getMethodName( kind, name ) ] = ( key ) => source.getEntityRecord( kind, name, key );
 	result[ getMethodName( kind, name, 'get', true ) ] = ( ...args ) => source.getEntityRecords( kind, name, ...args );
 	return result;
 }, {} );
 
+const createEntityRecordResolver = ( source ) => defaultEntities.reduce( ( result, entity ) => {
```

Comment:
These functions are incredibly dense and difficult to decipher. Could we make them more readable? Or alternatively, if it's a lost cause, could we at least include a function-level JSDoc describing intent of what the function is accomplishing?

Reply:
I'm starting to question whether these shortcuts are useful or if we just deprecated them in favor of the generic alternatives. Especially since they are very hard to statically analyze and document.

Reply:
I think they'll be useful once the shortcuts are documented.  It will be helpful in documentation to indicate the pattern used to generate the methodNames, that way developers can infer what shortcut to use when dispatching/selecting.  So not so much all the entity shortcuts would need documented, but just the pattern for the method names.